surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+
+- Linux (desktop or server)
+- Windows 10/11
+- Windows 8 or older
+- Mac OS
+- iOS
+- Android
+- Embedded platforms (with an operating system)
+- Embedded platforms (bare metal)
+- WebAssembly
+- Explicitly platform-independent (e.g., a library which does not interact with the operating system)
+- Other (open response)
+
+> **justification**
+>
+> This question can be used to figure out roughly what systems are being targeted as well as 
+> what OS stack is being developed against (i.e., desktop/server OS, mobile OS, embedded OS, bare metal)
+>
+> We're using "Linux" here rather than "*nix" or similar, with the same
+> justification as in the "Which operating systems do you use" question.
+>
+> We specifically care about the runtime environment being targeted. ISA and other machine specifics are
+> not what matters.
+
+### Which version(s) of Rust do you use for development (excluding CI and other automated testing)?
+
+Type: select all that apply (optional)
+
+- Current stable version
+- Previous stable version
+- A specific version of stable Rust equal to or newer than 1.67
+- A specific version of stable Rust older than 1.67
+- Beta release
+- Latest nightly
+- A specific version of nightly
+- Custom fork
+- I don't know
+- Other
+
+> **justification**
+>
+> Together with the following question, we can better determine what the spread of 
+> version usage is across the community.
+> We ask specifically about version 1.67 since it is, at the time of the survey, the 
+> version that was released ~1 year prior. Additionally, at the time of this writing
+> all major Linux distros have a version equal to or newer than this version.
+
+### If you use nightly for any tasks, why?
+
+Type: select all that apply (optional)
+
+- I don't use nightly
+- Out of habit
+- For a particular language feature or set of language features I need
+- I like to have access to all the latest features
+- To help test the nightly version for bugs
+- To provide design feedback on nightly features
+- For testing in CI
+- A crate dependency I use requires it
+- A tool I use requires it
+- Other (open response)
+
+> **justification**
+>
+> We'd like to know what are the common reasons people use nightly
+> so that we can better understand where testers are coming from.
+
+### What is the oldest version of Rust you use for any development task?
+
+Excluding testing to ensure your code works on that compiler version.
+
+Type: select one (optional)
+
+1.76 (nightly), 1.75 (nightly), and then every version from 1.74 to 1.0 in descending order, and "a pre-1.0 version"
+
+> **justification**
+>
+> To get real data on how many people use older versions of the toolchain to
+> inform discussion on MSRVs.
+
+### Which editor or IDE setup do you use with Rust code on a regular basis?
+
+Type: select all that apply (optional)
+
+- VS Code
+- vi/vim/neovim
+- IntelliJ/CLion/other JetBrains IDE + Rust plugin
+- Rust Rover (dedicated IntelliJ Rust IDE)
+- Emacs
```

Comment:
```suggestion
- Emacs (or derivative distributions like Doom Emacs, Spacemacs, etc.)
```

since we include `vim` and variants, let's play a fair game with fellow Emacs users? pretty please? :slightly_smiling_face:

Reply:
Done.